### PR TITLE
7097 isolate non UI tests

### PR DESCRIPTION
**Pull request by ngaruko**
# Description

Refactor those tests that do not require a browser (starting from e2e/api tests) to use the existing mocha/chai framework we have in integration tests and run them out of protractor

medic/cht-core #7097 

# Code review checklist

- [ ] Readable: Concise, well named, follows the [style guide](https://docs.communityhealthtoolkit.org/contribute/code/style-guide/), documented if necessary.
- [ ] Documented: Configuration and user documentation on [cht-docs](https://github.com/medic/cht-docs/)
- [ ] Tested: Unit and/or e2e where appropriate
- [ ] Internationalised: All user facing text
- [ ] Backwards compatible: Works with existing data and configuration or includes a migration. Any breaking changes documented in the release notes.

# License

The software is provided under AGPL-3.0. Contributions to this project are accepted under the same license.

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -157,7 +157,9 @@ jobs:
       run: |
         echo "COUCH_URL=$COUCH_URL HORTI_BUILDS_SERVER=$MARKET_URL/$BUILDS_SERVER"
         COUCH_URL=$COUCH_URL HORTI_BUILDS_SERVER=$MARKET_URL/$BUILDS_SERVER horti --local --install=medic:medic:test-$TRAVIS_BUILD_NUMBER > tests/logs/horti.log &
-    - name: Test it!
+    - name: Non UI Tests!
```
Comment (newtewt) [inline .github/workflows/build.yml:160]: I would rename this integration tests to hopefully encapsulate more! Also, I think this is worthy of it's own job.

File: tests/non-ui/config.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
+const request = require('request-promise-native');
```
Comment (newtewt) [inline tests/non-ui/config.js:1]: I like that this is pulled out into it's own module. Take the time to update `tests/conf.js` so we don't have duplicate code.

File: tests/non-ui/api/server.js [hunk 3]
```diff
@@ -14,15 +19,15 @@ describe('server', () => {
...
         .catch(e => {
-          expect(e.responseBody).toBe('Content-Type must be application/json');
+          expect(e.responseBody).to.equal('Content-Type must be application/json');
         });
     });
   });
 
   describe('response compression', () => {
-    afterAll(utils.afterEach);
+    after(async () => await utils.afterEachMocha());
```
Comment (newtewt) [inline tests/non-ui/api/server.js:30]: you can short hand here as well `after(utils.afterEachMocha);`

File: tests/non-ui/api/server.js [hunk 4]
```diff
@@ -1,10 +1,15 @@
...
+const auth = require('../../../tests/auth')();
 const request = require('request');
-const constants = require('../../constants');
+const constants = require('../../../tests/constants');
 const _ = require('lodash');
+const {expect} = require('chai');
+const config = require('../config');
 
 describe('server', () => {
+  before(async () => {
```
Comment (newtewt) [inline tests/non-ui/api/server.js:10]: you can short hand here `before(config.prepServices);`

**Review by newtewt — CHANGES_REQUESTED** [review]
I like this start to separate full e2e tests and integration tests. In that case can you rename these from non-ui to integration.

Can you also add to the TESTING.md on how to run these?

File: tests/non-ui/config.js (hunk 2 above)
Comment (ngaruko) [reply tests/non-ui/config.js:1]: good point - done

File: .github/workflows/build.yml (hunk 1 above)
Comment (ngaruko) [reply .github/workflows/build.yml:160]: I thought about calling them 'integration' but we have some `api/integration` tests, I wonder if that could get confusing ! Will try to make them a separate job while we decide on the nomenclature

**Pull request closed without merging**